### PR TITLE
Spaces to add comments

### PR DESCRIPTION
**Pull request by vinniebrice**

### CODE REVIEW

**Comment by vinniebrice** [comment]
Feel free to use this branch to make your changes

**Comment by vinniebrice** [comment]
First off, what level do we expect of someone doing this tutorial? There isn't much guidance so I assume intermediate? So after the first tutorial?

File: README.md [hunk 1]
```diff
@@ -2,16 +2,16 @@
  The GitHub API is documented here: https://docs.github.com/en/rest
 
  We will be connecting to this API to create a version of 'github'.
- You should clone this repo and in terminal type sbt run, in your browser on localhost:9000 you should see a page with the title "Welcome to your version of GitHub".
+ //You should clone this repo and in terminal type sbt run, in your browser on localhost:9000 you should see a page with the title "Welcome to your version of GitHub".
```
Comment (vinniebrice) [inline README.md:5]: Maybe provide the SSH key for them to clone?

File: README.md [hunk 2]
```diff
@@ -2,16 +2,16 @@
...
  
-  For package structure use standard: view (html templates), controller (API endpoint entry), service (Orchestration, should be light weight seeing as    we're only calling one endpoint), connector (To speak to api.github), model (Domain model data representation).
+  //For package structure use standard: view (html templates), controller (API endpoint entry), service (Orchestration, should be light weight seeing as    we're only calling one endpoint), connector (To speak to api.github), model (Domain model data representation).
```
Comment (vinniebrice) [inline README.md:7]: Could we maybe rephrase and ask them to create the packages views, controllers, ... and then explain what they mean after in bullet points? And maybe rephrase "(To speak to api.github)" to send requests to external services or something?

File: README.md [hunk 3]
```diff
@@ -2,16 +2,16 @@
...
  Using the GitHub API it’s a lot easier if upfront you start to use your own git credentials for authorisation (to avoid rate limits) -    https://developer.github.com/v3/auth/#basic-authentication (no need to worry about authorisation at the moment you can come back to this once you get to task 9 as authorisation is needed when you start creating and deleting in repositories on the github api)
 
- When writing your code only have working code on your main branch and for each new feature create a branch to create this. When this new feature is done and all the code is working create a pull request to merge your branch into main.
+ //When writing your code only have working code on your main branch and for each new feature create a branch to create this. When this new feature is done and all the code is working create a pull request to merge your branch into main.
```
Comment (vinniebrice) [inline README.md:10]: This is great practice and maybe we should further explain why? For example, why is it bad practice to push commits straight to main? And how do we know all the code is working per "feature". Whats our "definition of done"? All tests passing or a manual execution works?

File: README.md [hunk 4]
```diff
@@ -2,16 +2,16 @@
...
  
 ## Task 1.
  Your mongodb should contain users which will have the following information: username, date account created, location, number of followers, number following.
-    Look at the return of a user from the GitHub API to help you make your model of a user. Try inputting your GitHub username into the url to see: https://api.github.com/users/{yourUsername}
+    Look at the return of a user from the GitHub API to help you make your model of a user. //Try inputting your GitHub username into the url to see: https://api.github.com/users/{yourUsername}
```
Comment (vinniebrice) [inline README.md:14]: Were we supposed to add controller, service and connector methods by this point? Or are they provided?

File: README.md [hunk 5]
```diff
@@ -28,7 +28,7 @@
  Make routes for all the CRUD commands (Create Read Update Delete)
  
  ## Task 4.
-  Make a route to add a user from the GitHub API to your repository.
+  //Make a route to add a user from the GitHub API to your repository.
```
Comment (vinniebrice) [inline README.md:31]: Make a route... and add other methods?

File: README.md [hunk 6]
```diff
@@ -40,32 +40,32 @@
  On this page display the names of files & folders (as determined by an API call) for the selected usernames’ repository.
  
  ## Task 7.
-  Stretch exercise.
+  //Stretch exercise.
```
Comment (vinniebrice) [inline README.md:43]: What does "Stretch exercise" mean?

File: README.md [hunk 7]
```diff
@@ -40,32 +40,32 @@
...
   
-  Make a new route GET /github/users/:username/repos/:repoName/*path
+  //Make a new route GET /github/users/:username/repos/:repoName/*path
```
Comment (vinniebrice) [inline README.md:45]: Should we add a bit more guidance on how to break this large task up?

File: README.md [hunk 8]
```diff
@@ -40,32 +40,32 @@
...
   The path will be path to file I.E app/controllers/HelloWorldController.scala, you can base64 encode path for ease
   If the path given resolves to a file, display the plaintext of the file on the page.
   If the path given is a folder, display the files & folder names (In the same way it's displayed in the first part of Task 4)
   
 ## Task 8. 
- (Assuming task 7 stretch is complete)
+ //(Assuming task 7 stretch is complete)
```
Comment (vinniebrice) [inline README.md:51]: Probably not needed as all these tasks look like you can't move on without doing the one prior

File: README.md [hunk 9]
```diff
@@ -40,32 +40,32 @@
...
  Link up the pages. I.E the first step of task 5 doesn't only list the files/folders, but they are hyperlinks that will direct to the correct page in the  stretch exercise.
  For example, /github/users/jxr227/repos/ROScratch, will show various files and folders including the folder "exercises". If you click "exercises" you  will be directed to:
  /github/users/jxr227/repos/ROScratch/exercises
  
  Do the same on the /*path pages, so that you can, from the top level of a repository, browse all folders and files.
 
 ## Task 9.
- if you haven't already done so write integration and unit tests for all your methods. When it comes to unit testing your connector (the file using ws client that connects to the API) you can use wiremock to mock the API responses. 
+ //if you haven't already done so write integration and unit tests for all your methods. When it comes to unit testing your connector (the file using ws client that connects to the API) you can use wiremock to mock the API responses.
```
Comment (vinniebrice) [inline README.md:59]: This probably needs to come earlier to help stress test driven development? Up until now they have no way of knowing if what they have done is correct...

File: README.md [hunk 10]
```diff
@@ -40,32 +40,32 @@
...
 
 ## Task 9.
- if you haven't already done so write integration and unit tests for all your methods. When it comes to unit testing your connector (the file using ws client that connects to the API) you can use wiremock to mock the API responses. 
+ //if you haven't already done so write integration and unit tests for all your methods. When it comes to unit testing your connector (the file using ws client that connects to the API) you can use wiremock to mock the API responses. 
  
  To use wiremock in your library dependencies in the build.sbt file add:
     "com.github.tomakehurst" % "wiremock-jre8" % "2.33.2" % Test
 
- When you run test if you get an error about jackson databind after adding the wiremock try adding this to your build.sbt file:
+ //When you run test if you get an error about jackson databind after adding the wiremock try adding this to your build.sbt file:
```
Comment (vinniebrice) [inline README.md:64]: Replace "when" with "if"? Or add this to the repo they clone if its a known error?

File: README.md [hunk 11]
```diff
@@ -40,32 +40,32 @@
...
  dependencyOverrides +="com.fasterxml.jackson.core" % "jackson-databind" % "2.11.0"
 
 ## Task 10.
- Now try and get, create, update and delete files in a repository on the GitHub API.
+ //Now try and get, create, update and delete files in a repository on the GitHub API.
```
Comment (vinniebrice) [inline README.md:68]: Are they doing this via curl or with buttons/urls in their browser?

File: README.md [hunk 12]
```diff
@@ -76,5 +76,5 @@
  Now that you have accomplished step 10 instead of using curl requests in terminal try creating a form, so you can complete these routes from your browser.
 
 ## Task 12.
- Add some css styling to the layout of your pages. 
+ //Add some css styling to the layout of your pages.
```
Comment (vinniebrice) [inline README.md:79]: Could we provide an example?

File: README.md (hunk 6 above)
Comment (robynHM) [reply README.md:43]: parts i copied from James original post but you are right this is not needed

File: README.md (hunk 7 above)
Comment (robynHM) [reply README.md:45]: i enjoyed that after the book one this one was more difficult with not as much guidance and i had to figure stuff out. I don't want to give too much information. What did you have in mind for guidance?
